plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/core/ui/wizards/NewDockerConnectionPage.java

```diff
@@ -19,39 +19,41 @@
 import org.jboss.reddeer.jface.wizard.WizardPage;
 import org.jboss.reddeer.swt.api.Button;
 import org.jboss.reddeer.swt.api.Table;
+import org.jboss.reddeer.swt.condition.WidgetIsEnabled;
 import org.jboss.reddeer.swt.impl.button.CheckBox;
+import org.jboss.reddeer.swt.impl.button.FinishButton;
 import org.jboss.reddeer.swt.impl.button.PushButton;
 import org.jboss.reddeer.swt.impl.button.RadioButton;
 import org.jboss.reddeer.swt.impl.shell.DefaultShell;
 import org.jboss.reddeer.swt.impl.table.DefaultTable;
 import org.jboss.reddeer.swt.impl.text.LabeledText;
 import org.jboss.reddeer.swt.impl.toolbar.DefaultToolItem;
-import org.jboss.tools.docker.reddeer.ui.DockerExplorer;
+import org.jboss.tools.docker.reddeer.ui.DockerExplorerView;
 
 /**
  * 
- * @author jkopriva
+ * @author jkopriva@redhat.com
  *
  */
 
-public class DockerConnectionWizard extends WizardPage {
+public class NewDockerConnectionPage extends WizardPage {
+	private static final String NEW_DOCKER_CONNECTION_SHELL = "New Docker Connection";
 
-	public DockerConnectionWizard() {
+	public NewDockerConnectionPage() {
 		super();
 	}
 
 	public void open() {
-		DockerExplorer de = new DockerExplorer();
-		de.open();
+		new DockerExplorerView().open();
 		new DefaultToolItem("Add Connection").click();
-		new WaitUntil(new ShellWithTextIsAvailable("New Docker Connection"));
+		new WaitUntil(new ShellWithTextIsAvailable(NEW_DOCKER_CONNECTION_SHELL));
 	}
 
 	public void finish() {
 		String shell = new DefaultShell().getText();
```

Comment:
Why do you retrieve text of currently focused shell? What if there is a modal shell, some popup etc. Then this would be broken. Does the New docker connection wizard change its title while it is opened? If not, then you can remove this and on line 58 where you are using this text you can use constant again.